src/assets/Keycloak/Keycloak.md

```diff
@@ -0,0 +1,148 @@
+      KC_DB: postgres
+      KC_DB_URL: "jdbc:postgresql://postgres:5432/keycloak"
+      KC_DB_USERNAME: keycloak
+      KC_DB_PASSWORD: keycloak
+      KC_FEATURES: impersonation
+      KEYCLOAK_ADMIN: admin
+      KEYCLOAK_ADMIN_PASSWORD: admin
+    ports:
+      - 8001:8080 # map keycloak port 8080 to 8001
+    depends_on:
+      - postgres
+
+volumes:
+  postgres_data:
+```
+
+Save the above code in `compose.yaml` file and run `docker compose up` in a terminal.
+
+Docker compose serves Keycloak on `http://localhost:8001`.
+
+### By Distribution File
+There is another way to set up Keycloak using the Keycloak distribution file.
+1. Download the Keycloak distribution
+    ```
+    export KC_VERSION=24.0.4
+    curl -LO  https://github.com/keycloak/keycloak/releases/download/"${KC_VERSION}"/keycloak-"${KC_VERSION}".zip
+    ```
+
+2. Unzip the Keycloak distribution package
+    ```
+    unzip keycloak-${KC_VERSION}.zip
+    ```
+
+3. Install openjdk
+    ```
+    sudo apt install openjdk-17-jdk
+    ```
+
+4. Navigate to the Keycloak directory
+    ```
+    cd keycloak-${KC_VERSION}
+   ```
+
+5. When we start the server for the first time, we have to set the admin user and the admin password:
+    ```
+   KEYCLOAK_ADMIN=admin KEYCLOAK_ADMIN_PASSWORD=admin ./bin/kc.sh start-dev
+   ```
+    > Note: if port 8080 is already used, then you can map the Keycloak port using the Keycloak environment variable 'KC_HTTP_PORT=8001'
+
+6. When we start again, it is not necessary to set these variables again. You can start the server with:
+    ```
+   ./bin/kc.sh start-dev
+   ```
+   > Note: start-dev runs the Keycloak application in DEV-mode. Do not use this for production.
+
+By default, the Keycloak server is on port 8080 for http and 8084 for https. They are only served from the localhost loopback address 127.0.0.1:
+Keycloak serves on `http://localhost:<port>`.
+
+## Why Keycloak Matters?
+1. Keycloak is open source.
+2. Keycloak supports three different authentication protocols which gives you the possibility to cover many applications with different security demands with a single tool.
+3. Keycloak provides a web-based GUI which makes any configurations changes easy.
+4. Keycloak has huge [community support](https://www.keycloak.org/community).
+
+## Keycloak Drawbacks
+Despite its extensive features, Keycloak does have certain limitations. One of these is the need for a more varied implementation approach.
+Additionally, the following are some of the drawbacks associated with Keycloak:
+1. Complex Server Deployment:
+   The manual process involved in deploying Keycloak on a server can be complicated, potentially affecting overall productivity.
+
+2. Lack of Comprehensive Support Documentation:
+   Despite its robust functionalities, Keycloak could benefit from more efficient and extensive support documentation. This would make it easier for users to find suitable solutions to their problems.
+
+## When It May Not Be the Best Choice?
```

Comment:
```suggestion
## When It May Not Be the Best Choice
```

similar here - this is a "statement" heading, not a question, even though it starts with "When"